test/service/test_request_validation.py

```diff
@@ -119,6 +142,27 @@ def test_bad_sort_facet_but_valid_filter_facet_of_specimen(self):
         self.assertEqual(400, response.status_code, response.json())
         self.assertEqual(self.sort_facet_message, response.json())
 
+    def test_file(self):
+        url = self.base_url + "repository/files?from=1&size=1"
+        response = requests.get(url)
+        self.assertEqual(200, response.status_code, response.content)
+
+    def test_valid_filter_facet_of_file(self):
+        url = self.base_url + "repository/files?from=1&size=1&filters={'file':{'organ':{'is':['fake-val']}}}"
+        response = requests.get(url)
+        self.assertEqual(200, response.status_code, response.content)
+
+    def test_valid_sort_facet_of_file(self):
+        url = self.base_url + "repository/specimens?size=15&sort=organ&order=asc"
+        response = requests.get(url)
+        self.assertEqual(200, response.status_code, response.content)
+
+    def test_valid_sort_and_filter_facet_of_file(self):
+        url = self.base_url + "repository/specimens?size=15&sort=organ&order=asc" \
+                              "&filters={'file':{'organ':{'is':['fake-val2']}}}"
+        response = requests.get(url)
+        self.assertEqual(200, response.status_code, response.content)
+
```

Comment:
If you mean the  data source of the metadata, that would be the ElasticSearch Docker container. Currently, every test case inherits from this class.  Below is the code for that class: https://github.com/DataBiosphere/azul/blob/9f11f5ca3f8b5b0c68d9bd887e2d2686a6c7301f/test/es_test_case.py#L14-L66.

The docker container is populated using fake data generated by this class: https://github.com/DataBiosphere/azul/blob/9f11f5ca3f8b5b0c68d9bd887e2d2686a6c7301f/test/service/data_generator/fake_data_utils.py#L46-L80

And the random data has a schema defined by this template file. 
https://github.com/DataBiosphere/azul/blob/develop/test/service/data_generator/fake_data_template.json

Again the data is randomized, so I can't hard code the test values and use them to spot-check the response content. I need to create a new data loader that loads metadata that is explicitly defined and not randomized.

Reply:
Thanks for the clear and through description of the test input data, @GPelayo.
So, the input JSON values are random, but you could still compare the expected response dictionary structure and keys with the expected structure and keys. This would be sufficient for me to have reasonable confidence that the code was working as expected.
Yet, I just can't see approving a PR for what is intended to be a functional test that only checks the return code.
There are some tips/tricks regarding python dict key comparison [here](https://stackoverflow.com/questions/13714982/comparing-two-dictionaries-key-values-and-returning-the-value-if-match)  that may be helpful, although for our purposes, I expect it would need to be a recursive comparison, as I suspect  the response includes nested objects.
Please consider comparing the actual and expect responses wrt their structure and keys.